lib/decorators/party_name_attributes.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'scraped'
+
+# Members on the members page are grouped by party but the names or ids of the
+# parties are not given. Instead, each party is signified by an image.
+
+# This decorator maps the party name and id to the image name and populates the
+# party image <img> tags with party_name and party_id attributes.
+class PartyNameAttributes < Scraped::Response::Decorator
+  PARTY_IMAGES_MAPPED =
+    {
+      'pri01.png'                     => ['Partido Revolucionario Institucional', 'PRI'],
+      'pan.png'                       => ['Partido Acción Nacional', 'PAN'],
+      'prd01.png'                     => ['Partido de la Revolución Democrática', 'PRD'],
+      'logvrd.jpg'                    => ['Partido Verde Ecologista', 'PVE'],
+      'logo_movimiento_ciudadano.png' => ['Movimiento Ciudadano', 'MC'],
+      'logpt.jpg'                     => ['Partido del Trabajo', 'PT'],
+      'panal.gif'                     => ['Partido Nueva Alianza', 'PANAL'],
+      'LogoMorena.jpg'                => ['Movimiento Regeneración Nacional', 'MORENA'],
+      'encuentro.png'                 => ['Encuentro Social', 'PES'],
+      'logo_SP.jpg'                   => ['Sin Partido', '_IND'],
+      'independiente.png'             => ['Sin Partido', '_IND'],
+    }.freeze
+
+  def body
+    Nokogiri::HTML(super).tap do |doc|
+      doc.css('img').each do |img|
+        p = party_name_and_id(img)
+        next if p.empty?
+        img[:party_name] = p.values.first.first
+        img[:party_id] = p.values.first.last
+      end
+    end.to_s
+  end
+
+  private
+
+  def party_name_and_id(img)
+    PARTY_IMAGES_MAPPED.select { |k, _v| img[:src].include? k }
```

Comment:
If you're only ever calling `.first` on the return value of this, then you'd be better testing with `.find` rather than `.select`. But you'd be much much better still treating the hash as a hash, and doing a direct lookup, rather than iterating over it.

Though